content/labs/lab10/index.en.md

```diff
@@ -0,0 +1,162 @@
+
+You must implement the following computation parallelization methods:
+
+- `MultiThreadGenerator`: A multi-threaded method that manually creates and manages `Thread` objects.
+- `TasksGenerator`: A method that parallelizes the work using the thread pool (`ThreadPool`) via `Task` objects.
+- `ParallelGenerator`: A method that uses the TPL (Task Parallel Library).
+
+`Program.cs` contains the logic for measuring time and running each generator sequentially.
+
+A correctly generated fractal should look as follows:
+
+<div style="text-align: center;">
+  <img src="/labs/lab10/mandelbrotset.png" width="400px" alt="mandelbrotset.png" />
+</div>
+
+{{% hint warning %}}
+**Implementation Notes**
+
+- **Number of threads:**
+  - In the `MultiThreadGenerator` and `TasksGenerator` implementations, you should create `N` units of work (threads/tasks), where `N` is equal to the number of processor cores. This is the optimal number for 100% CPU-bound tasks.
+- **Work division:**
+  - When generating the fractal, the simplest strategy is to divide the image into `N` equal, horizontal strips.
+  - Calculate how many rows fall to one thread, and then in a loop, pass the appropriate range to each thread/task for processing.
+
+{{% /hint %}}
+
+{{% hint info %}}
+**Helpful Materials:**
+
+- [Microsoft Learn: Threads and threading](https://learn.microsoft.com/en-us/dotnet/standard/threading/threads-and-threading)
+- [Microsoft Learn: Task Class](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-9.0)
+- [Microsoft Learn: Write a Simple Parallel.For Loop](https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/how-to-write-a-simple-parallel-for-loop)
+
+{{% /hint %}}
+
+### Example Solution
+
+> [!TIP]
+> **Solution**
+> {{< filetree dir="labs/lab10/solution/FractalsGenerator" >}}
+
+## Offer Aggregator
+
+{{% hint info %}}
+**What is asynchronous programming?**
+
+Asynchronous programming is used to avoid blocking the execution thread while waiting for operations that are outside our control.
+We use it for tasks such as waiting for a response from a network (API), sending queries to a database, or reading the contents of a large file from disk.
+
+In this case, the goal is not faster computation, but better management of waiting time and application responsiveness.
+
+{{% /hint %}}
+
+### Task Description
+
+The goal of the task is to build an asynchronous console client (in the `FlightScanner.Client` project) that will aggregate flight offers from multiple services simultaneously. The client will communicate with a locally running REST API (the `FlightScanner.API` project).
+
+The API, running at `http://localhost:5222` (the address can be found in the [launchSettings.json](/labs/lab10/solution/FlightScanner.API/Properties/launchSettings.json) file), is fully implemented and does not require modification. It must be run before running the console application.
+
```

Comment:
Na Linuxie nie ma runtime'a asp.neta. Napisz może żeby odpalali to przez dostarczony przez nas skrypt run.sh
```suggestion

> [!NOTE]
> On Linux run the `FlightScanner.API` project through the provided script `run.sh`. It will download missing `ASP.NET` runtime, then run the project.

```

I wklej srypt do solution/FlightScanner.API/run.sh i odpowiednio kodu startowego student/FlightScanner.API/run.sh:
```
#!/bin/sh
wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
chmod +x dotnet-install.sh
./dotnet-install.sh --channel 9.0
./dotnet-install.sh --channel 9.0 --runtime aspnetcore
export PATH=/home/$USER/.dotnet:$PATH
dotnet run
```